File: datasets/health/medical-data/modify/profile_activity.swift
import SwiftUI
struct profile_activity: View {
    @State private  var gender_answer  : String?
    var body: some View {
        ScrollView
        {
            ZStack
            {
                VStack
                {
                    VStack
                    {
                        Text("NAME")
                            .padding(.top,20)
                        TextField("", text: .constant(""))
                    }
                    Text("EMAIL")
                        .padding(.top,30)
                    Text("BIRTH DATE")
                        .padding(.top,30)
              
                    Text("GENDER")
                        .padding(.top,30)
                    HStack
                    {
                        Button(action: { gender_answer = "MALE" })
                        {
                            HStack
                            {
                                Image(systemName: gender_answer == "MALE" ? "checkmark.circle.fill" : "circle")
                                Text("MALE")
                                    .foregroundColor(.black)
                            }
                        }
                        Button(action: { gender_answer = "FEMALE" })
                        {
                            HStack
                            {
                                Image(systemName: gender_answer == "FEMALE" ? "checkmark.circle.fill" : "circle")
                                    .padding(.leading,50)
                                    .padding(.leading,50)
                                Text("FEMALE")
                                    .foregroundColor(.black)
                                    .padding(.leading,50)
                                    .padding(.leading,50)
                            }
                            .padding(.leading,50)
                            .padding(.leading,50)
                        }
                        .padding(.leading,50)
                        .padding(.leading,50)
                    }
                    Text("INTRODUCE YOUR PIN")
                        .padding(.top,23)
                    Button(action: {})
                    {
                        Text("SAVE CHANGES")
                            .foregroundColor(.black)
                    }
                    .background(Color.white)
                    .padding(.top,40)
                    .padding(.bottom,60)
                }
                .padding(.top,16)
                .padding(.bottom,16)
                .padding(.leading,16)
                .padding(.trailing,16)
            }
        }}}

struct profile_activity_Previews: PreviewProvider {
    static var previews: some View {
        profile_activity()
    }
}

File: experiments/health/medical-data/compose/profile_activity.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun profile_activity() {
  Column(modifier = Modifier.fillMaxWidth().fillMaxHeight().verticalScroll(rememberScrollState())) {
    Box(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
      Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(start = 16.0.dp).padding(end = 16.0.dp).padding(top = 16.0.dp).padding(bottom = 16.0.dp), verticalArrangement = Arrangement.Top, horizontalAlignment = Alignment.Start) {
        Text(text = "NAME", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(top = 20.0.dp))
        OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.wrapContentWidth().wrapContentHeight())
        Text(text = "EMAIL", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(top = 30.0.dp))
        OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.wrapContentWidth().wrapContentHeight())
        Text(text = "BIRTH DATE", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(top = 30.0.dp))
        Column(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
        }
        Text(text = "GENDER", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(top = 30.0.dp))
        Row(modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(top = 12.0.dp)) {
          Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
            /* TODO: bind selection state */
            RadioButton(selected = false, onClick = {})
            Text("MALE")
          }
          Row(modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 50.0.dp)) {
            /* TODO: bind selection state */
            RadioButton(selected = false, onClick = {})
            Text("FEMALE")
          }
        }
        Text(text = "INTRODUCE YOUR PIN", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(top = 23.0.dp))
        OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.wrapContentWidth().wrapContentHeight())
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(top = 40.0.dp).padding(bottom = 60.0.dp)) {
          Text(text = "SAVE CHANGES")
        }
      }
    }
  }
}


@Preview(showBackground = true)
@Composable
fun profile_activity_preview() {
    MyApplicationTheme {
        profile_activity()
    }
}
